sklearn/utils/metaestimators.py

```diff
@@ -32,7 +37,13 @@ def __get__(self, obj, type=None):
         if obj is not None:
             # delegate only on instances, not the classes.
             # this is to allow access to the docstrings.
-            self.get_attribute(obj)
+            for item in self.delegate_names:
+                if hasattr(obj, item):
```

Comment:
If the point of using attrgetter before was to allow the delegate to have '.' in it, then we'd better maintain that, with something like:
```python
try:
    delegate = attrgetter(delegate_name)(obj)
except AttributeError:
    continue
else:
    getattr(delegate, self.method_name)
    break
```